### ISSUE TITLE
Not working with wire:navigate

### ISSUE THREAD

**Issue by QualityCodeSA** [aliqasemzadeh/livewire-bootstrap-modal#13]
Bootstrapt modal working fine when first time open page with refresh but in livewire 3 not working when come back from other page (usinf wire:navigate) and open modal

**Comment by aliqasemzadeh** [comment]
yes,
I know some problem and I am working hard on it.

**Comment by QualityCodeSA** [comment]
> yes, I know some problem and I am working hard on it.

Thank you brother. Update me once finish so I will apply changes

**Comment by aliqasemzadeh** [comment]
Check note and update.
I hope this update help me and then let me know.

**Comment by QualityCodeSA** [comment]
@aliqasemzadeh any update regarding not working with wire:navigate ?

**Comment by aliqasemzadeh** [comment]
@QualityCodeSA 
No it doesn't work.
I think we need trait to dispatch modals to solve this problem.

**Comment by aliqasemzadeh** [comment]
@QualityCodeSA 
Trait doesn't work.

**Comment by QualityCodeSA** [comment]
@aliqasemzadeh any possibility to work or not?

**Comment by aliqasemzadeh** [comment]
@QualityCodeSA 
It looks like there is no solution a the moment.

**Comment by mozhuilungdsuo** [comment]
I think we need to persist the livewire modal? @persist('modals')

<livewire-modals 

@endpersist
@aliqasemzadeh @QualityCodeSA

**Comment by aliqasemzadeh** [comment]
@mozhuilungdsuo 
Please give us example.

**Comment by mozhuilungdsuo** [comment]
as per the documentation a 
`<a href="/users" wire:navigate>Users</a>` 
 requests the page in the background and replaces the current page. So the livewire modal component ie
`<livewire:modals/>` we put in the app layout view would show the old livewire component and since we added a new livewire component  itll give issue ( thats what i assume). so as per the documentation here[https://livewire.laravel.com/docs/navigate](url)

we should be keeping this in the app layout view. `
@persist('player')
    <livewire:modals/>
@endpersist`
I maybe wrong but this should ideally fix it.

**Comment by r7zq** [comment]
@mozhuilungdsuo yeah persist works, another issue i found is modal-backdrop not being created so the solution i found is to add event listener for livewire:navigated and dispose the old modal
`document.addEventListener("livewire:navigated", () => {
        let modal = Modal.getInstance(modalsElement);
        if (modal) {
            modal.dispose();
        }
    });`
